src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1230,7 +1237,7 @@ CopyLocalDataIntoShards(Oid distributedRelationId)
 
 	/* get the table columns */
 	tupleDescriptor = RelationGetDescr(distributedRelation);
-	slot = MakeSingleTupleTableSlot(tupleDescriptor);
+	slot = MakeSingleTupleTableSlotCompat(tupleDescriptor, &TTSOpsHeapTuple);
```

Comment:
Do we need `TTSOpsHeapTuple ` or is it OK to use `TTSOpsMinimalTuple `?

Would you mind describing the difference between both?

Reply:
`ExecStoreHeapTuple` errors if it doesn't get `TTSOpsHeapTuple`